conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -183,6 +186,20 @@ private static void frameworkException(HttpServerExchange exchange, FrameworkExc
         writeResponse(exchange, Optional.of(ConjureError.fromServiceException(exception)), statusCode);
     }
 
+    private static void deadlineExpiredException(HttpServerExchange exchange, DeadlineExpiredException exception) {
+        // need to further narrow down to external versus internal exception types
+        if (exception instanceof DeadlineExpiredException.External externalException) {
+            log.info("external deadline exceeded", externalException);
+            writeResponse(exchange, Optional.empty(), 400);
+        } else if (exception instanceof DeadlineExpiredException.Internal internalException) {
+            log.error("internal deadline exceeded", internalException);
+            writeResponse(exchange, Optional.empty(), 500);
+        } else {
+            log.error("unknown deadline exception type", exception);
+            writeResponse(exchange, Optional.empty(), 500);
```

Comment:
perhaps it would be better to fall back to the `ServiceException` case instead here?

Reply:
Ya, the full response with errorInstanceId would be helpful